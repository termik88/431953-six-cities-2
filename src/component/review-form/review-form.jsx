import React from "react";
import PropTypes from "prop-types";

import ReviewFormStar from "../review-form-star/review-form-star.jsx";

export const stars = [
  {number: 5, title: `perfect`},
  {number: 4, title: `good`},
  {number: 3, title: `not bad`},
  {number: 2, title: `badly`},
  {number: 1, title: `terribly`},
];

const ReviewForm = ({isActive, review, isValidationReviewForm, handleInputChange, handleSendComment, isLoading, errorInfo}) => {
  return (
    <form
      className="reviews__form form"
      action="#"
      method="post">
      <label className="reviews__label form__label" htmlFor="review">Your review</label>
      <div className="reviews__rating-form form__rating">
        {stars.map((star) => (
          <ReviewFormStar
            key = {`key-star-${star.number}-${isActive}`}
            star = {star}
            handleInputChange = {handleInputChange}
            isLoading = {isLoading}
          />))}
      </div>
      <textarea
        onChange={handleInputChange}
        value={review}
        className="reviews__textarea form__textarea"
        id="review"
        name="review"
        placeholder="Tell how was your stay, what you like and what can be improved"
        disabled={isLoading}/>
      <div className="reviews__button-wrapper">
        <p className="reviews__help">
          To submit review please make sure to set <span className="reviews__star">rating</span> and
          describe your stay with at least <b className="reviews__text-amount">50 characters</b>.
        </p>
        <button
          onClick={handleSendComment}
          className="reviews__submit form__submit button"
          type="submit"
          disabled={!isValidationReviewForm || isLoading}>Submit</button>
      </div>
      { errorInfo && <span style={{color: `red`}}>{errorInfo}</span> }
    </form>
  );
};

ReviewForm.propTypes = {
  review: PropTypes.string.isRequired,
  isActive: PropTypes.node.isRequired,
  isValidationReviewForm: PropTypes.bool.isRequired,
  isLoading: PropTypes.bool.isRequired,
  errorInfo: PropTypes.string.isRequired,
  handleInputChange: PropTypes.func.isRequired,
  handleSendComment: PropTypes.func.isRequired
};

export default ReviewForm;
