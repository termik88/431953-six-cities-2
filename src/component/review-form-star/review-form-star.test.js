import React from 'react';
import renderer from 'react-test-renderer';

import ReviewFormStar from './review-form-star.jsx';
import {stars} from "../review-form/review-form.jsx";

describe(`ReviewFormStar component test.`, () => {
  it(`Correctly renders after relaunch`, () => {
    const mockDate = {
      star: stars[0],
      handleInputChange: jest.fn(),
      isLoading: false
    };

    const test = renderer
      .create(
          <ReviewFormStar
            {...mockDate}
          />)
      .toJSON();

    expect(test).toMatchSnapshot();
  });
});
