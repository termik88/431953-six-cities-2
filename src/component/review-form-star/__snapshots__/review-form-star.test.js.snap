// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewFormStar component test. Correctly renders after relaunch 1`] = `
Array [
  <input
    className="form__rating-input visually-hidden"
    disabled={false}
    id="5-stars"
    name="rating"
    onChange={[MockFunction]}
    type="radio"
    value={5}
  />,
  <label
    className="reviews__rating-label form__rating-label"
    htmlFor="5-stars"
    title="perfect"
  >
    <svg
      className="form__star-image"
      height="33"
      width="37"
    >
      <use
        xlinkHref="#icon-star"
      />
    </svg>
  </label>,
]
`;
